Sources/FPSStatusBarViewController.swift

```diff
@@ -86,18 +86,21 @@ internal class FPSStatusBarViewController: UIViewController, FPSCounterDelegate
 
     // MARK: - Getting the shared status bar window
 
+    static var previouslyCreatedStatusBarWindow: UIWindow?
```

Comment:
I don't think this extra variable is needed here, as it's essentially the same as the `statusBarWindow` var. That one is created on initialization and stored, so you can just access it directly.

Reply:
Well, the thing is that we need to check it without instantiation (see the first guard in hide()). If you reference lazy var directly, afaik, it will be initialized (hence window will be created), and that is not the thing you want when you (accidentally) invoke `hide()` without `show()`?

Reply:
Since `statusBarWindow` is not declared lazy it will get instantiated immediately afaik. We could of course add the lazy attribute, but I feel for this kind of project that's overkill (after all, it's just a debugging tool).

In general I would assume that hiding the counter would imply that you intend to show it at some point at least, so creating it eagerly (if maybe unnecessarily) is not an issue.

Reply:
Well, syntax-wise it's not lazy, but given that it's *static* (Swift) variable, it will be lazily initialized at the point of first reference.

Quoting https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/Properties.html :

> Stored type properties are lazily initialized on their first access. They are guaranteed to be initialized only once, even when accessed by multiple threads simultaneously, and they do not need to be marked with the lazy modifier.

So we of course could keep `hide()` without this guard, and technically it would *probably* work (today), but I just do feel like it's kind of misleading due to this side effect (window creation on lazy initialization) there. In my opinion it's the worst of all evils (as whether it works or not depends on the way window is initialized). Another possible variant would be tracking int with a flag and/or asserting it without guarding - imho anything is better than just assuming that it will work as is. Anyway, it's up to you to decide - it's a minor thing imho, but I just don't want to leave it without an assert/guard/etc..

Reply:
Thanks for the reference, I did not know that about the static vars 👍

Regardless wether the instantiation is immediate or as in this case lazy though, it's not a question of wether it works or not. Even lazy variables are guaranteed to be initialized before they're first accessed. Therefore I guess this boils down to a potential inefficiency (creating a window which is not going to be used).

While I do see your point, in this specific case I would prefer having the inefficiency (since this is hopefully not going to be used in a production build) over a (slightly) more complex solution.